File: csharp/Gherkin/Ast/Step.cs
using System;
using System.Linq;

namespace Gherkin.Ast
{
	public class Step : IHasLocation
	{
		public Location Location { get; private set; }
		public string Keyword { get; private set; }
		public string Value { get; private set; }
		public StepArgument StepArgument { get; private set; }

		public Step(string keyword, string value, StepArgument stepArgument, Location location)
		{
			Location = location;
			Keyword = keyword;
			Value = value;
			StepArgument = stepArgument;
		}
	}
}

File: src/main/java/gherkin/ast/Step.java
package gherkin.ast;

public class Step extends Node implements DescribesItself {
    private final String keyword;
    private final String name;
    private final StepArgument argument;

    public Step(Location location, String keyword, String name, StepArgument argument) {
        super(location);
        this.keyword = keyword;
        this.name = name;
        this.argument = argument;
    }

    public String getName() {
        return name;
    }

    public StepArgument getArgument() {
        return argument;
    }

    public String getKeyword() {
        return keyword;
    }

    @Override
    public void describeTo(Visitor visitor) {
        visitor.visitStep(this);
    }
}
